### DISCUSSION TITLE
HTMX was never an option for me.

### DISCUSSION THREAD

**Discussion by marcodpt** [bigskysoftware/htmx#3054] [General]
`HTMX` is a fantastic idea, and the [essays page](https://htmx.org/essays/) is a class apart.

I highly recommend reading all the articles.

However, I believe that the entire page should be shared via `link`, making it possible to reconstruct the `state` through your `link`.

This means that every page transformation using the server must add parameters to the `URL query`.

And in the case of transformations in the frontend, you must use the `URL hash` to save the `state` of the app.

In this case I believe that a `hash router` that reacts to hash changes is the correct way to add dynamic javascript to the page while being able to share `links` with the `state`.

This is why `HTMX` was never an option for me.

I would like to know your opinion and please correct me if there is a way to use `HTMX` in this approach or if someone has done something similar.

**Comment by bennadel** [comment]
I'm not sure that I understand what you are saying. But, as I am coming out of the AngularJS world, one thing that I am trying to translate in my head is how to use the URL to drive Modal / Fly-outs while still also being able to **refresh** the page to return to the same state. One thought that I has was to use the `hash` to manage the modal or any kind of "secondary layer" state. But, right now, HTMX doesn't seem to expose the ability to tap into the `history` API directly. I mean, I know I can use the `history` API myself; but, I don't think it will work properly with all the page-caching/restoration if I do that.

All to say, there doesn't seem to be a super obvious way to have a multi-layer app that _also_ handles page-refresh and history. I'm still trying to work it out in my head.
